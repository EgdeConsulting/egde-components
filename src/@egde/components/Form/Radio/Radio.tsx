import React, { ReactElement } from 'react';
import { Radio as ChakraRadio, RadioGroup, Stack } from '@chakra-ui/react';
import { RadioProps } from '@egde/types';
import { BaseInput } from '@egde/components';

export const Radio = (props: RadioProps): ReactElement => {
    const {
        direction = 'column',
        options,
        value,
        onChange,
        invalidText,
        isDisabled,
        backgroundColor,
    } = props;

    return (
        <BaseInput {...props}>
            <RadioGroup
                isDisabled={isDisabled}
                onChange={(e: any) => onChange(+e)}
                value={value}
                backgroundColor={backgroundColor}
            >
                <Stack direction={direction}>
                    {options.map((option, index) => {
                        return (
                            <ChakraRadio
                                isInvalid={!!invalidText}
                                key={`rd${index}`}
                                value={index}
                            >
                                {option}
                            </ChakraRadio>
                        );
                    })}
                </Stack>
            </RadioGroup>
        </BaseInput>
    );
};
