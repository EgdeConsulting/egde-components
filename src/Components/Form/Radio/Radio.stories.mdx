import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { DocsPageHeader, DocsPageBody } from 'Docs';
import { LibPath } from 'Types';
import { Radio } from 'Components';
import { useState } from 'react';

export const Template = (args) => {
    const [value, setValue] = useState();
    const handleChange = (e) => {
        setValue(+e);
    };
    return (
        <div>
            <Radio {...args} value={value} onChange={handleChange} />
        </div>
    );
};

export const componentName = 'Radio';

<Meta
    title={LibPath.Form + componentName}
    component={Radio}
    args={{
        options: ['Option 1', 'Option 2', 'Option 3'],
        value: 0,
        onChange: () => {},
        isDisabled: false,
        backgroundColor: '#FFFFFF',
        direction: 'column',
        autoFocus: false,
        label: 'Title',
        labelTextMargin: '10px',
        invalidText: '',
        captionText: 'Caption text',
        captionTextMargin: '10px',
        margin: '0',
        width: '100%',
        minWidth: '0',
        maxWidth: '100%',
    }}
/>

<DocsPageHeader
    componentName={componentName}
    componentDescription="This component is used to create a checklist where only one answers can be selected."
    addImportantNote={false}
    variant="stable"
/>

<Canvas withToolbar>
    <Story name="Base">{Template.bind({})}</Story>
</Canvas>

<DocsPageBody addStories={true} />

### Horizontal layout

<Canvas withToolbar>
    <Story
        name="Horizontal layout"
        args={{
            direction: 'row',
            captionText: '',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Invaild Input

<Canvas withToolbar>
    <Story
        name="Invaild Input"
        args={{
            captionText: '',
            backgroundColor: '#F9E5E5',
            width: '30%',
            invalidText: 'Please select an option!',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>
