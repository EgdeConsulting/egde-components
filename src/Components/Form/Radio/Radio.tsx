import React, { ReactElement } from 'react';
import {
    FormLabel,
    Radio as ChakraRadio,
    RadioGroup,
    Stack,
} from '@chakra-ui/react';
import { RadioProps } from 'Types';
import { BaseInput } from 'Components';

export function Radio(props: RadioProps): ReactElement {
    const {
        label,
        labelTextMargin,
        direction = 'column',
        options,
        value,
        onChange,
        invalidText,
        captionText,
        isDisabled,
        margin,
        padding,
        width,
        minWidth,
        backgroundColor,
    } = props;

    return (
        <BaseInput
            invalidText={invalidText}
            captionText={captionText}
            margin={margin}
            padding={padding}
            width={width}
            minWidth={minWidth}
            isGroup
        >
            <FormLabel
                fontWeight='bold'
                color='darkgrey'
                margin={labelTextMargin || '0 0 8px 0'}
                as='legend'
            >
                {label}
            </FormLabel>
            <RadioGroup
                isDisabled={isDisabled}
                onChange={(e) => onChange(+e)}
                value={value}
                backgroundColor={backgroundColor}
            >
                <Stack direction={direction}>
                    {options.map((option, index) => {
                        return (
                            <ChakraRadio
                                isInvalid={!!invalidText}
                                key={`rd${index}`}
                                value={index}
                            >
                                {option}
                            </ChakraRadio>
                        );
                    })}
                </Stack>
            </RadioGroup>
        </BaseInput>
    );
}
