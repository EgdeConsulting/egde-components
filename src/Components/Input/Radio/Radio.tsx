import { ReactElement } from 'react';
import { Radio as ChakraRadio, RadioGroup, Stack } from '@chakra-ui/react';
import { RadioProps } from 'Types';
import { BaseInput } from '../BaseInput'
import React from 'react';

export function Radio(props: RadioProps): ReactElement {
    const {
        direction = 'column',
        options,
        value,
        onChange,
        invalidText,
        isDisabled,
        backgroundColor,
    } = props;

    return (
        <BaseInput {...props}>
            <RadioGroup
                isDisabled={isDisabled}
                onChange={(e) => onChange(+e)}
                value={value}
                backgroundColor={backgroundColor}
            >
                <Stack direction={direction}>
                    {options.map((option, index) => {
                        return (
                            <ChakraRadio
                                isInvalid={!!invalidText}
                                key={`rd${index}`}
                                value={index}
                            >
                                {option}
                            </ChakraRadio>
                        );
                    })}
                </Stack>
            </RadioGroup>
        </BaseInput>
    );
}
