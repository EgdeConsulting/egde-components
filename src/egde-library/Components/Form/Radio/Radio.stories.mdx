import { Meta, Story, Canvas } from '@storybook/addon-docs';
import { LibPath, DocsPageHeader, DocsPageBodyComponents } from 'egde-library';
import { Radio } from 'egde-components';
import { useState } from 'react';

export const Template = (args) => {
    const [value, setValue] = useState();
    const handleChange = (e) => {
        setValue(+e);
    };
    return (
        <div>
            <Radio {...args} value={value} onChange={handleChange} />
        </div>
    );
};

export const componentName = 'Radio';

<Meta
    title={LibPath.Form + componentName}
    component={Radio}
    args={{
        options: ['Option 1', 'Option 2', 'Option 3'],
        value: 0,
        onChange: () => {},
        isDisabled: false,
        backgroundColor: '#FFFFFF',
        direction: 'column',
        autoFocus: false,
        label: 'Title',
        labelTextMargin: '10px',
        invalidText: '',
        captionText: 'Caption text',
        captionTextMargin: '10px',
        margin: '0',
        width: '100%',
        minWidth: '0',
        maxWidth: '100%',
    }}
/>

<DocsPageHeader
    title={componentName}
    description="This component is used to create a checklist where only one answers can be selected."
    addImportantNote={false}
    variant="stable"
/>

<Canvas withToolbar>
    <Story name="Base">{Template.bind({})}</Story>
</Canvas>

<DocsPageBodyComponents addStories={true} />

### Horizontal layout

<Canvas withToolbar>
    <Story
        name="Horizontal layout"
        args={{
            direction: 'row',
            captionText: '',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>

### Invaild Input

<Canvas withToolbar>
    <Story
        name="Invaild Input"
        args={{
            captionText: '',
            backgroundColor: '#F9E5E5',
            width: '30%',
            invalidText: 'Please select an option!',
        }}
    >
        {Template.bind({})}
    </Story>
</Canvas>
