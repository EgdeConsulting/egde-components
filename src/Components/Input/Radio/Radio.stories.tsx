import { ComponentStory, ComponentMeta } from '@storybook/react';
import React from 'react';
import { Radio } from './Radio';
import { LibPath } from 'Types';

export default {
    title: LibPath.Form + 'Radio' || 'undefined',
    component: Radio,

    argTypes: {
        options: {defaultValue: [
            'option 1',
            'option 2',
            'option 3',
        ]},
        value: {defaultValue: 0},
        label: {defaultValue: 'This is a Radio component'}
    }
} as unknown as ComponentMeta<typeof Radio>;

const Template: ComponentStory<typeof Radio> = (args) => (
  <Radio {...args} />
);

export const RadioStd = Template.bind({});


