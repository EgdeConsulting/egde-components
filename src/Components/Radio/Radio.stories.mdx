import { Meta, Story, Props, Canvas, Preview } from '@storybook/addon-docs/blocks';
import { Radio } from './Radio';
import { LibPath } from 'Types';

export const Template = (args) => <Radio {...args} />;

<Meta
  title={LibPath.Form + 'Radio'}
  component={Radio}
  parameters={{
    toolbar: {
      title: { hidden: true },
      // zoom: { hidden: true },
      // eject: { hidden: true },
      // copy: { hidden: true },
      // fullscreen: { hidden: true },
      // 'storybook/background': { hidden: true },
      // 'storybook/viewport': { hidden: true },
    },
  }}
  args={{
    options: ['option 1', 'option 2', 'option 3'],
    value: 0,
    onChange: () => {},
    isDisabled: false,
    backgroundColor: '#fff',
    direction: 'column',
    autoFocus: false,
    label: 'Radio',
    labelTextMargin: '0',
    invalidText: '',
    captionText: '',
    captionTextMargin: '0',
    margin: '0',
    padding: '0',
    width: '100%',
    minWidth: '0',
    maxWidth: '100%',
  }}
/>

# Radio

<div
  style={{
    borderRadius: '999999px',
    display: 'inline-block',
    padding: '2px 8px',
    fontSize: '14px',
    lineHeight: '20px',
    fontWeight: '700',
    textAlign: 'center',
    letterSpacing: '0.25px',
    backgroundColor: '#138849',
    color: '#ffffff',
    marginBottom: '16px',
    marginRight: '3px',
  }}
>
  Stable
</div>

<div
  style={{
    borderRadius: '999999px',
    display: 'inline-block',
    padding: '2px 8px',
    fontSize: '14px',
    lineHeight: '20px',
    fontWeight: '700',
    textAlign: 'center',
    letterSpacing: '0.25px',
    backgroundColor: '#D2CB3F',
    color: '#ffffff',
    marginBottom: '16px',
    marginRight: '3px',
  }}
>
  Unstable
</div>

<div
  style={{
    borderRadius: '999999px',
    display: 'inline-block',
    padding: '2px 8px',
    fontSize: '14px',
    lineHeight: '20px',
    fontWeight: '700',
    textAlign: 'center',
    letterSpacing: '0.25px',
    backgroundColor: '#D23F47',
    color: '#ffffff',
    marginBottom: '16px',
    marginRight: '3px',
  }}
>
  Deprecated
</div>

<hr />

<Preview withToolbar>
  <Story name="Radio">{Template.bind({})}</Story>
</Preview>

<Props story="Radio" />

# Stories
