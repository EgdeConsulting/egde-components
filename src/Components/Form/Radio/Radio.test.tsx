import React from 'react';
import { render } from '@testing-library/react';
import { Radio } from './Radio';



// Snapshot Test
describe('Snapshot Radio', () => {
  it('should render correctly', () => {
    const tree = render(<Radio
        options= {['option 1', 'option 2', 'option 3']}
        value= {0}
        onChange= {() => {}}
        isDisabled= {false}
        backgroundColor= '#fff'
        direction= 'column'
        autoFocus= {false}
        label= 'Radio'
        labelTextMargin= '0'
        invalidText= ''
        captionText= ''
        captionTextMargin= '0'
        margin= '0'
        padding= '0'
        width= '100%'
        minWidth= '0'
        maxWidth= '100%'
        />
        );
    expect(tree).toMatchSnapshot();
  });
});


