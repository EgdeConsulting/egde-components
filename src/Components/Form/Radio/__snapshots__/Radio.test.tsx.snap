// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio component test should render correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div
      id="chakra-toast-portal"
    >
      <ul
        id="chakra-toast-manager-top"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-top-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-top-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom-left"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
      <ul
        id="chakra-toast-manager-bottom"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column; margin: 0px auto;"
      />
      <ul
        id="chakra-toast-manager-bottom-right"
        style="position: fixed; z-index: 5500; pointer-events: none; display: flex; flex-direction: column;"
      />
    </div>
    <div>
      <div
        class="css-k456he"
      >
        <p
          class="chakra-text css-154u26z"
        >
          Radio label
        </p>
        <div
          class="css-1o2srd0"
        >
          <div
            class="chakra-radio-group css-1k9ek97"
            role="radiogroup"
          >
            <div
              class="chakra-stack css-n21gh5"
            >
              <label
                class="chakra-radio css-1pw4d56"
              >
                <input
                  class="chakra-radio__input"
                  id="radio-1"
                  name="radio-3"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="radio"
                  value="0"
                />
                <span
                  aria-hidden="true"
                  class="chakra-radio__control css-ssalds"
                />
                <span
                  class="chakra-radio__label css-1y8kf23"
                >
                  First Radio Option
                </span>
              </label>
              <label
                class="chakra-radio css-1pw4d56"
              >
                <input
                  class="chakra-radio__input"
                  id="radio-2"
                  name="radio-3"
                  style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                  type="radio"
                  value="1"
                />
                <span
                  aria-hidden="true"
                  class="chakra-radio__control css-ssalds"
                />
                <span
                  class="chakra-radio__label css-1y8kf23"
                >
                  Last Radio Option
                </span>
              </label>
            </div>
          </div>
        </div>
        <p
          class="chakra-text css-1vrfh4l"
        >
          Caption text
        </p>
        
      </div>
      <button
        type="button"
      >
        Toggle disabled
      </button>
      <button
        type="button"
      >
        Invalidate
      </button>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-k456he"
    >
      <p
        class="chakra-text css-154u26z"
      >
        Radio label
      </p>
      <div
        class="css-1o2srd0"
      >
        <div
          class="chakra-radio-group css-1k9ek97"
          role="radiogroup"
        >
          <div
            class="chakra-stack css-n21gh5"
          >
            <label
              class="chakra-radio css-1pw4d56"
            >
              <input
                class="chakra-radio__input"
                id="radio-1"
                name="radio-3"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="radio"
                value="0"
              />
              <span
                aria-hidden="true"
                class="chakra-radio__control css-ssalds"
              />
              <span
                class="chakra-radio__label css-1y8kf23"
              >
                First Radio Option
              </span>
            </label>
            <label
              class="chakra-radio css-1pw4d56"
            >
              <input
                class="chakra-radio__input"
                id="radio-2"
                name="radio-3"
                style="border: 0px; clip: rect(0px, 0px, 0px, 0px); height: 1px; width: 1px; margin: -1px; padding: 0px; overflow: hidden; white-space: nowrap; position: absolute;"
                type="radio"
                value="1"
              />
              <span
                aria-hidden="true"
                class="chakra-radio__control css-ssalds"
              />
              <span
                class="chakra-radio__label css-1y8kf23"
              >
                Last Radio Option
              </span>
            </label>
          </div>
        </div>
      </div>
      <p
        class="chakra-text css-1vrfh4l"
      >
        Caption text
      </p>
      
    </div>
    <button
      type="button"
    >
      Toggle disabled
    </button>
    <button
      type="button"
    >
      Invalidate
    </button>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
